### ISSUE TITLE
Spreadsheet: wrong cell size

### ISSUE THREAD

**Issue by hdmr14** [ONLYOFFICE/DocumentServer#2942]
### This issue is unique.

- [X] I have used the [search tool](https://github.com/ONLYOFFICE/DocumentServer/issues?q=) and did not find an issue describing my bug.

### Operating System of DocumentServer

Docker

### Version information

8.0.2.143

### Expected Behavior

The cell size should be set and displayed in the same way as in Excel or other compatible editors.

### Actual Behavior

If the cell size is specified as a 50x50px square in Excel or other compatible editors, it appears as 29x50px in ONLYOFFICE Spreadsheet.
When setting the cell size to 50x50px in ONLYOFFICE Spreadsheet, the cell is clearly not square, and when opened in Excel, it becomes 58x50px. Additionally, the size (aspect ratio) of the images inserted in the table also changes to match this strange behavior of the cell dimensions.

### Reproduction Steps

1. Change the cell size to 50x50 in Excel.  
2. Open it in ONLYOFFICE Spreadsheet.  
3. In ONLYOFFICE Spreadsheet, it is displayed as 29x50px. At this point, the 50x50px image is also distorted accordingly.

### Additional information

This issue also occurs in the Desktop version.

**Comment by sushiltechnex** [comment]
maybe this solution will solve your problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by VladimirProkofiev52** [comment]
Hello! @hdmr14 
I was unable to reproduce your issue. Could you please let me know which version of Excel you are using? If possible, please provide the file where the issue occurs.

**Comment by hdmr14** [comment]
@VladimirProkofiev52 Thanks for your comment! I am using Microsoft® Excel® 2019 MSO (version 2410 build 16.0.18129.20100) 64-bit.
The attached file is a 50x50px table created in Excel 2019. 

- [50x50-by-excel.xlsx](https://github.com/user-attachments/files/17681729/50x50-by-excel.xlsx)

The image below is screenshots of this file displayed in Excel, LibreOffice Calc, Google Spreadsheets, and ONLYOFFICE Spreadsheet.

![image](https://github.com/user-attachments/assets/cc5ebd62-a6ca-4fe0-9397-ab21cce1c86a)

While investigating for this report, I noticed that Google Spreadsheets also fails to reproduce the correct cell size, similar to ONLYOFFICE Spreadsheet. I don't have a version of Excel newer than 2019, but I have confirmed that cell sizes display correctly in Excel for Microsoft 365, and that the issue persists even when the file is saved in Microsoft 365 Excel.

**Comment by hdmr14** [comment]
Upon further investigation, it appears that most Office-compatible software attempts to replicate the display behavior of Excel. However, ONLYOFFICE Spreadsheet and Google Spreadsheets significantly deviate from maintaining the original aspect ratio. Unless ONLYOFFICE Spreadsheet is specifically designed to follow Google Spreadsheets as its reference, this behavior can reasonably be considered a bug.

While OpenOffice also shows a slight discrepancy, it seems to be caused by a different issue compared to ONLYOFFICE Spreadsheet and Google Spreadsheets.

Additionally, although I am not sure if this is related, I conducted these tests in a Japanese-language environment.

![comparison](https://github.com/user-attachments/assets/5ad6c841-5791-4daf-86ca-1015baece757)

**Comment by Dmitry-Kov** [comment]
Hi @hdmr14,

I created ticket #71775 with your issue. When some fixes will be done we'll text you here in this issue.
Thank you for the report!

**Comment by hdmr14** [comment]
I investigated the cause of this issue further.

The root cause is upstream in font resolution and column-width calculation. The drawing distortion is only a downstream effect of the resulting incorrect cell geometry.

The workbook[(50x50-by-excel.xlsx)](https://github.com/user-attachments/files/17681729/50x50-by-excel.xlsx)’s built-in Normal style uses:

```text
name="游ゴシック" (Yu Gothic)
scheme="minor"
charset=128
```

The workbook theme defines:

```text
minorFont.latin = Calibri
minorFont[Jpan] = 游ゴシック
```

ONLYOFFICE initially parses the original font information correctly. However, during the [`core` to `sdkjs` binary conversion](https://github.com/ONLYOFFICE/core/blob/3250a848ee4ef20c2fb8c38dc86350ec579124b8/OOXML/Binary/Sheets/Reader/BinaryWriterS.cpp#L1675-L1725), `BinaryStyleTableWriter::WriteFont` obtains the RFont value through [`FontProcessor::getFont`](https://github.com/ONLYOFFICE/core/blob/3250a848ee4ef20c2fb8c38dc86350ec579124b8/OOXML/Binary/Document/DocWrapper/FontProcessor.cpp#L93-L148).

When a font has `scheme="minor"`, `FontProcessor::getFont` [returns the theme’s `minorFont.latin` value at this branch](https://github.com/ONLYOFFICE/core/blob/3250a848ee4ef20c2fb8c38dc86350ec579124b8/OOXML/Binary/Document/DocWrapper/FontProcessor.cpp#L103-L104), instead of using the original font name.

In this workbook, `minorFont.latin` is Calibri, so the RFont written to the binary stream becomes Calibri rather than the original Yu Gothic font.

On the `sdkjs` side, [`Binary_StylesTableReader.ReadFonts`](https://github.com/ONLYOFFICE/sdkjs/blob/72b0421c0bbf9d01eed9cf14834ae47eb2df1b50/cell/model/Serialize.js#L9284-L9302) reads the font data and calls [`Font.prototype.checkSchemeFont`](https://github.com/ONLYOFFICE/sdkjs/blob/72b0421c0bbf9d01eed9cf14834ae47eb2df1b50/cell/model/WorkbookElems.js#L1034-L1046), which assigns the major or minor Latin theme font to `font.fn` again.

As a result, when ONLYOFFICE [`_calcMaxDigitWidth`](https://github.com/ONLYOFFICE/sdkjs/blob/72b0421c0bbf9d01eed9cf14834ae47eb2df1b50/cell/view/WorkbookView.js#L4464-L4496) measures the maximum width of the digits 0–9 for the default column width, it uses Calibri instead of the Yu Gothic Normal font name recorded in the workbook.

This produces narrower columns than when Yu Gothic is used for the measurement.

The row height is imported through a separate path and remains unchanged. The image uses a two-cell anchor, so its displayed size is derived from the resulting cell grid. Because only the columns become narrower, the image is compressed horizontally.

In short:

```text
Normal font: Yu Gothic
    ↓
scheme="minor" is resolved as minorFont.latin
    ↓
Calibri is used for digit-width measurement
    ↓
columns become too narrow
    ↓
the two-cell anchored image inherits the narrow grid
    ↓
the image is horizontally distorted
```

I reproduced the same relative behavior with a minimized workbook in ONLYOFFICE Desktop Editors 9.4.0.129.

**Comment by Dmitry-Kov** [comment]
Hi @hdmr14,

Thank you for taking the time to dig into this — the breakdown is very helpful.

I've passed your analysis to the developers, along with the confirmation that the
issue still reproduces in 9.4.0.129.

Could you also attach the minimized workbook you used for that check? It would help
the developers verify the font resolution path faster.

We'll also check whether the same happens with other East Asian charsets, since the
scheme="minor" resolution looks charset-independent.

We'll update you here once there's progress.

**Comment by hdmr14** [comment]
@Rita-Bubnova, Thank you for the update.

I've attached [`main-regression.xlsx`](https://github.com/user-attachments/files/30784877/main-regression.xlsx), which is the exact minimized fixture I used for the ONLYOFFICE Desktop Editors 9.4.0.129 check.

A note about its origin: this is not a copy of the original workbook attached to the Issue. It is an anonymous, deterministically generated fixture derived from an Excel-native geometry baseline. The baseline was created independently in Excel with:

* 25 pt rows
* four equal-width columns
* an A1:E5 two-cell anchor
* a square shape

The fixture then retains only the font/theme state relevant to this issue:

```text
Normal font:
  name="游ゴシック" (Yu Gothic)
  scheme="minor"
  charset=128

Theme:
  minorFont.latin = Calibri
  minorFont[Jpan] = 游ゴシック
```

The workbook does not contain personal data, external links, macros, embedded objects, media, or embedded font files.

### Expected result

In Excel 16.0, the workbook opens with:

```text
Columns A:D: 24.857 pt each
Rows:         25 pt
Shape:        99.429 × 99.429 pt
Anchor:       A1:E5
```

The four-column by four-row grid and the anchored shape are therefore square.

### Result in ONLYOFFICE Desktop Editors 9.4.0.129

Using the same file:

```text
Column A: 3.43 symbols
Row 1:    25 pt
Shape:    3.08 × 3.51 cm
Aspect:   0.877
Anchor:   A1:E5
```

The row height and two-cell anchor remain unchanged, but the columns become narrower. The shape inherits that narrower grid and is therefore compressed horizontally.

So the expected behavior is not a drawing-specific adjustment. The expected behavior is for the column geometry to be calculated using the correct Normal-font metrics, which should preserve the approximately square grid and shape.

> We'll also check whether the same happens with other East Asian charsets, since the
> scheme="minor" resolution looks charset-independent.

I agree that similar effects may occur with other non-Latin scripts, since this theme-font resolution path appears to select the Latin font independently of the charset.

**Comment by Dmitry-Kov** [comment]
@hdmr14,

Got the fixture, thanks — it's been passed on to the developers along with your analysis.

I verified the geometry on our side and the numbers match, so they have everything
they need. We'll update you here once there's progress.
